lib/Extension/LanguageServer/Protocol/CompletionItem.php

```diff
@@ -0,0 +1,206 @@
+     */
+    public $insertTextFormat;
+
+    /**
+     * An edit which is applied to a document when selecting this completion. When an edit is provided the value of
+     * `insertText` is ignored.
+     *
+     * *Note:* The range of the edit must be a single line range and it must contain the position at which completion
+     * has been requested.
+     *
+     * @var TextEdit|null
+     */
+    public $textEdit;
+
+    /**
+     * An optional array of additional text edits that are applied when
+     * selecting this completion. Edits must not overlap (including the same insert position)
+     * with the main edit nor with themselves.
+     *
+     * Additional text edits should be used to change text unrelated to the current cursor position
+     * (for example adding an import statement at the top of the file if the completion item will
+     * insert an unqualified type).
+     *
+     * @var TextEdit[]
+     */
+    public $additionalTextEdits = [];
+
+    /** 
+     * An optional set of characters that when pressed while this completion
+     * is active will accept it first and then type that character. *Note* that
+     * all commit characters should have `length=1` and that superfluous
+     * characters will be ignored.
+     *
+     * @var string[]
+     */
+    public $commitCharacters = [];
+
+    /**
+     * An optional command that is executed *after* inserting this completion. *Note* that
+     * additional modifications to the current document should be described with the
+     * additionalTextEdits-property.
+     *
+     * @var Command|null
+     */
+    public $command;
+
+    /**
+     * An data entry field that is preserved on a completion item between
+     * a completion and a completion resolve request.
+     *
+     * @var mixed
+     */
+    public $data;
+
+    public function __construct(
+        string $label,
+        int $kind = null,
+        string $detail = null,
+        string $documentation = null,
+        bool $deprecated = false,
+        bool $preselect = false,
+        string $sortText = null,
+        string $filterText = null,
+        string $insertText = null,
+        string $insertTextFormat = null,
+        TextEdit $textEdit = null,
+        array $additionalTextEdits = [],
+        array $commitCharacters = [],
+        Command $command = null,
+        $data = null
+    )
+    {
+        $this->label = $label;
+        $this->kind = $kind;
+        $this->detail = $detail;
+        $this->documentation = $documentation;
+        $this->deprecated = $deprecated;
+        $this->preselect = $preselect;
+        $this->sortText = $sortText;
+        $this->filterText = $filterText;
+        $this->insertText = $insertText;
+        $this->insertTextFormat = $insertTextFormat;
+        $this->textEdit = $textEdit;
+        $this->additionalTextEdits = $additionalTextEdits;
+        $this->commitCharacters = $commitCharacters;
+        $this->command = $command;
+        $this->data = $data;
```

Comment:
Can probably remove all the optional parameters here and depend only on properties.